src/zarr/store/remote.py

```diff
@@ -51,18 +52,22 @@ def __init__(
         """
 
         super().__init__(mode=mode)
-
         if isinstance(url, str):
-            self._fs, self.path = fsspec.url_to_fs(url, **storage_options)
+            self._url = url.rstrip("/")
+            self._fs, _path = fsspec.url_to_fs(url, **storage_options)
```

Comment:
Is there a case where url has no trailing "/", but the return fom url_to_fs does?

Reply:
I am not aware of any, but I wanted to ensure that the UPath and non-UPath code paths do the exact same thing to the url string.

Reply:
I might be misunderstanding the question...